### PR TITLE
feature(canvas) drag metastrategy

### PR DESCRIPTION
**Pull request by enidemi**
I'm creating this metastrategy that collects base dragging strategies together to solve the following problem:

1. there are cases where no dragging is available and there is no visual feedback but if you keep dragging this element when reaching the parents edge it triggers a reparent strategy
3. it's possible that the escape hatch is automatically activated without user input in these cases, the goal of this metastrategy to be an alternative for that with a higher fitness

Today this has effect on multiline flex and flow layouts where the reorder is restricted but there are available reparent strategies when leaving the parent.

The fallback strategy is only shown in the strategy picker when there are no reparent or real move/drag strategies.

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/strategies/move-reorder-metastrategy.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+// + ancestor strategy !!???
+const baseMoveStrategyFactories: Array<CanvasStrategyFactory> = [
+  absoluteMoveStrategy,
+  flexReorderStrategy,
+  flowReorderStrategy,
+  relativeMoveStrategy,
+]
+
+export const moveReorderMetaStrategy: MetaCanvasStrategy = (
```
Comment (balazsbajorics) [inline editor/src/components/canvas/canvas-strategies/strategies/move-reorder-metastrategy.tsx:33]: it sounds like this is all drag-to-move strategies: move, reorder, reparent. I don't think this is too much at all! actually, I think this is a pretty elegant solution.

```suggestion
export const dragToMoveMetaStrategy: MetaCanvasStrategy = (
```

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/strategies/drag-to-move-metastrategy.tsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+export function doNothingStrategy(
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession | null,
+  customStrategyState: CustomStrategyState,
+): CanvasStrategy {
+  const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+  return {
+    id: 'DO_NOTHING',
+    name: 'Placeholder',
```
Comment (gbalint) [inline editor/src/components/canvas/canvas-strategies/strategies/drag-to-move-metastrategy.tsx:82]: I think we should find a better name..... Placeholder is pretty cryptic.

Comment (enidemi) [reply editor/src/components/canvas/canvas-strategies/strategies/drag-to-move-metastrategy.tsx:82]: oh yes, maybe 'Move' would fit better :)

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/strategies/drag-to-move-metastrategy.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    interactionSession.interactionData.modifiers.alt
+  ) {
+    return []
+  }
+
+  const ancestorStrategies = lookForApplicableParentStrategy(
+    canvasState,
+    interactionSession,
+    customStrategyState,
+  )
```
Comment (Rheeseyb) [inline editor/src/components/canvas/canvas-strategies/strategies/drag-to-move-metastrategy.tsx:56]: This can now be removed

**Pull request merged**